### PR TITLE
Improve service indicator initialization and add tests for under/overflow

### PR DESCRIPTION
**Pull request by andrewhop**
### Issues:
Addresses CryptoAlg-907

### Description of changes: 
Add lazy initialization logic to all service indicator functions. This is *not* an issue for the current Linux FIPS branch. This issue only happens on Windows FIPS builds. The Windows issue is nothing was calling `FIPS_service_indicator_update_state` which created the state before calling `FIPS_service_indicator_lock_state`. When lock was called with a null state nothing happens and AWS-LC continues and eventually hits a call to update state which creates the state, then something calls unlock and underflows. My hypothesis is MSVC changes the order of self tests in the `BORINGSSL_bcm_power_on_self_test` to hit this unlock issue while Clang/GCC does not.

### Call-outs:
The CI does not test FIPS on Windows yet due to other issues. I have tested this part on a Windows machine manually and it addressed the issue. On Linux there is no behavior change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: crypto/fipsmodule/service_indicator/service_indicator.c [hunk 1]
```diff
@@ -19,39 +19,36 @@ int is_fips_build(void) {
...
+    indicator = OPENSSL_malloc(sizeof(struct fips_service_indicator_state));
+    if (indicator == NULL) {
+      OPENSSL_PUT_ERROR(CRYPTO, ERR_R_MALLOC_FAILURE);
+      return NULL;
+    }
+    indicator->lock_state = STATE_UNLOCKED;
+    indicator->counter = 0;
+    if (!CRYPTO_set_thread_local(AWSLC_THREAD_LOCAL_FIPS_SERVICE_INDICATOR_STATE, indicator, OPENSSL_free)) {
+      OPENSSL_PUT_ERROR(CRYPTO, ERR_R_INTERNAL_ERROR);
+      return NULL;
```
Comment (torben-hansen) [inline crypto/fipsmodule/service_indicator/service_indicator.c:37]: Memory leak here (`indicator`)?

File: crypto/fipsmodule/service_indicator/service_indicator_test.cc [hunk 2]
```diff
@@ -664,7 +665,11 @@ struct AEADTestVector {
         kAESKey, 16, kAESCCMCiphertext, 64, AWSLC_APPROVED, false },
 };
 
-class AEAD_ServiceIndicatorTest : public testing::TestWithParam<AEADTestVector> {};
+class AEAD_ServiceIndicatorTest : public testing::TestWithParam<AEADTestVector> {
```
Comment (torben-hansen) [inline crypto/fipsmodule/service_indicator/service_indicator_test.cc:668]: Possible to move this destructor to a base class and derive all these other algorithm-specific service indicator classes from it?

**Review by torben-hansen — APPROVED** [review]

**Review by samuel40791765 — APPROVED** [review]
